tests/glsl-intrinsic/intrinsic-texture.slang

```diff
@@ -1,25 +1,15 @@
 #version 450 core
 #extension GL_EXT_texture_shadow_lod : enable
 
-//TEST:SIMPLE(filecheck=CHECK_GLSL): -allow-glsl -stage compute -entry computeMain -target glsl -DCOMPUTE
-//TEST:SIMPLE(filecheck=CHECK_GLSL): -allow-glsl -stage fragment -entry computeMain -target glsl
-//TEST:SIMPLE(filecheck=CHECK_SPV):  -allow-glsl -stage fragment -entry computeMain -target spirv -emit-spirv-via-glsl
-//TEST:SIMPLE(filecheck=CHECK_SPV_DIRECT): -allow-glsl -stage compute -entry computeMain -target spirv -DCOMPUTE
-//TEST:SIMPLE(filecheck=CHECK_SPV_DIRECT): -allow-glsl -stage fragment -entry computeMain -target spirv
-//TEST:SIMPLE(filecheck=CHECK_CUDA): -allow-glsl -stage compute -entry computeMain -target cuda -DCOMPUTE
-//TEST:SIMPLE(filecheck=CHECK_CUDA): -allow-glsl -stage fragment -entry computeMain -target cuda
-
-// Disabling following targets because they are currently causing compile errors.
-//DISABLE_TEST:SIMPLE(filecheck=CHECK_HLSL): -allow-glsl -stage compute -entry computeMain -target hlsl -DCOMPUTE
-//DISABLE_TEST:SIMPLE(filecheck=CHECK_CPP):  -allow-glsl -stage compute -entry computeMain -target cpp -DCOMPUTE
-
-// It appears that CUDA doesn't have implementations for multisampling samplers.
-// When targetting glsl and spirv, the DEBUG build of slangc encounters issues with "Array" variant
-// of multisampling samplers: sampler2DMSArray, isampler2DMSArray and usampler2DMSArray.
-//#define TEST_when_multisampling_works
-
-//RWStructuredBuffer<float> outputBuffer;
-//layout(location = 0) out vec4 o_color;
+//TEST:SIMPLE(filecheck=HLSL): -allow-glsl -stage compute  -entry computeMain -target hlsl -DCOMPUTE
```

Comment:
may be worth adding compute derivative tests for hlsl/glsl implicit LOD functions under the `-DCOMPUTE` tests.

Syntax is in `tests\glsl-intrinsic\compute-derivative\intrinsic-derivative-function-in-compute.slang`

Reply:
Thx for the pointer. I will check it out

Reply:
Were you referring to this test?
```
bool derivativeFunc()
{
    return true
        && dFdx(1.0f) != -1.0f
        && dFdy(1.0f) != -1.0f
        && dFdxFine(1.0f) != -1.0f
        && dFdyFine(1.0f) != -1.0f
        && dFdxCoarse(1.0f) != -1.0f
        && dFdyCoarse(1.0f) != -1.0f
        && fwidth(1.0f) != -1.0f
        && fwidthFine(1.0f) != -1.0f
        && fwidthCoarse(1.0f) != -1.0f
        ;
}
```
These are not texture functions.
If needed, they should be in a separate test file.
This test is already too big.

Reply:
I refer to:
```
        && int(0) == textureSize(gsampler1D, int(0))
        && vec2(0) == textureQueryLod(gsampler1D, float(0))
        && int(0) == textureQueryLevels(gsampler1D)
        && gvec4(T(0)) == texture(gsampler1D, float(0))
        && gvec4(T(0)) == texture(gsampler1D, float(0), float(0))
        && gvec4(T(0)) == textureProj(gsampler1D, vec2(0))
        && gvec4(T(0)) == textureProj(gsampler1D, vec2(0), float(0))
        && gvec4(T(0)) == textureProj(gsampler1D, vec4(0))
        && gvec4(T(0)) == textureProj(gsampler1D, vec4(0), float(0))
        && gvec4(T(0)) == textureLod(gsampler1D, float(0), float(0))
        && gvec4(T(0)) == textureOffset(gsampler1D, float(0), __LINE__)
        && gvec4(T(0)) == textureOffset(gsampler1D, float(0), __LINE__, float(0))
        && gvec4(T(0)) == texelFetch(gsampler1D, int(0), int(0))
        && gvec4(T(0)) == texelFetchOffset(gsampler1D, int(0), int(0), __LINE__)
        && gvec4(T(0)) == textureProjOffset(gsampler1D, vec2(0), __LINE__)
        && gvec4(T(0)) == textureProjOffset(gsampler1D, vec2(0), __LINE__, float(0))
        && gvec4(T(0)) == textureProjOffset(gsampler1D, vec4(0), __LINE__)
        && gvec4(T(0)) == textureProjOffset(gsampler1D, vec4(0), __LINE__,float(0))
        && gvec4(T(0)) == textureLodOffset(gsampler1D, float(0), float(0), __LINE__)
        && gvec4(T(0)) == textureProjLod(gsampler1D, vec2(0), float(0))
        && gvec4(T(0)) == textureProjLod(gsampler1D, vec4(0), float(0))
        && gvec4(T(0)) == textureProjLodOffset(gsampler1D, vec2(0), float(0), __LINE__)
        && gvec4(T(0)) == textureProjLodOffset(gsampler1D, vec4(0), float(0), __LINE__)
        && gvec4(T(0)) == textureGrad(gsampler1D, float(0), float(0), float(0))
        && gvec4(T(0)) == textureGradOffset(gsampler1D, float(0), float(0), float(0), __LINE__)
        && gvec4(T(0)) == textureProjGrad(gsampler1D, vec2(0), float(0), float(0))
        && gvec4(T(0)) == textureProjGrad(gsampler1D, vec4(0), float(0), float(0))
        && gvec4(T(0)) == textureProjGradOffset(gsampler1D, vec2(0), float(0), float(0), __LINE__)
        && gvec4(T(0)) == textureProjGradOffset(gsampler1D, vec4(0), float(0), float(0), __LINE__)
        && vec4(0) == texture1D(uniform_sampler1D, float(0))
        && vec4(0) == texture1D(uniform_sampler1D, float(0), float(0))
        && vec4(0) == texture1DProj(uniform_sampler1D, vec2(0))
        && vec4(0) == texture1DProj(uniform_sampler1D, vec2(0), float(0))
        && vec4(0) == texture1DProj(uniform_sampler1D, vec4(0))
        && vec4(0) == texture1DProj(uniform_sampler1D, vec4(0), float(0))
        && vec4(0) == texture1DLod(uniform_sampler1D, float(0), float(0))
        && vec4(0) == texture1DProjLod(uniform_sampler1D, vec2(0), float(0))
        && vec4(0) == texture1DProjLod(uniform_sampler1D, vec4(0), float(0))
```

Reply:
these are in the same test.
the test tests for derivatives in compute shader.

Implicit LOD functions need a derivative. NV hardware has a method to get a derivative from NV hardware

Reply:
Right.
Since this test is already too big, I will create a new test for the `__requireComputeDerivative()`; or I may modify the existing one, `intrinsic-derivative-function-in-compute.slang`.

Reply:
I think modify the existing one